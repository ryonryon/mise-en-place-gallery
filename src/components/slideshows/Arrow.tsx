import * as React from "react";

interface Props {
  height: string;
  width: string;
}

export const LeftArrow = ({ height, width }: Props) => (
  <svg width={width} height={height} viewBox="0 0 50 80" xmlSpace="preserve">
    <polyline
      fill="none"
      stroke="#FFFFFF"
      strokeWidth="1"
      strokeLinecap="round"
      strokeLinejoin="round"
      points="45.63,75.8 0.375,38.087 45.63,0.375"
    />
  </svg>
);

LeftArrow.defaultProps = {
  height: "30px",
  width: "30px"
};

export const RightArrow = ({ height, width }: Props) => (
  <svg width={width} height={height} viewBox="0 0 50 80" xmlSpace="preserve">
    <polyline
      fill="none"
      stroke="#FFFFFF"
      strokeWidth="1"
      strokeLinecap="round"
      strokeLinejoin="round"
      points="0.375,0.375 45.63,38.087 0.375,75.8"
    />
  </svg>
);

RightArrow.defaultProps = {
  height: "30px",
  width: "30px"
};
