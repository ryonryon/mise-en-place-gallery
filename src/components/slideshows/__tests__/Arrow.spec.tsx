import * as React from "react";
import * as renderer from "react-test-renderer";

import { LeftArrow, RightArrow } from "../Arrow";

describe("Arrow test", () => {
  it("tests Arrow render correctly", () => {
    const leftArrow = renderer.create(<LeftArrow />);
    const rightArrow = renderer.create(<RightArrow />);

    expect(leftArrow).toMatchSnapshot();
    expect(rightArrow).toMatchSnapshot();
  });
});
