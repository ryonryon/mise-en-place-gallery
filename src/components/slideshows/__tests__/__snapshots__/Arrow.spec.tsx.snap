// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Arrow test tests Arrow render correctly 1`] = `
<svg
  height="30px"
  viewBox="0 0 50 80"
  width="30px"
  xmlSpace="preserve"
>
  <polyline
    fill="none"
    points="45.63,75.8 0.375,38.087 45.63,0.375"
    stroke="#FFFFFF"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="1"
  />
</svg>
`;

exports[`Arrow test tests Arrow render correctly 2`] = `
<svg
  height="30px"
  viewBox="0 0 50 80"
  width="30px"
  xmlSpace="preserve"
>
  <polyline
    fill="none"
    points="0.375,0.375 45.63,38.087 0.375,75.8"
    stroke="#FFFFFF"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth="1"
  />
</svg>
`;
